internal/internal_workflow_client.go

```diff
@@ -1887,6 +1887,7 @@ func (w *workflowClientInterceptor) PollWorkflowUpdate(
 		cancel()
 		if err == context.DeadlineExceeded ||
 			status.Code(err) == codes.DeadlineExceeded ||
+			serviceerror.ToStatus(err).Code() == codes.DeadlineExceeded ||
 			(err == nil && resp.GetOutcome() == nil) {
```

Comment:
I was under the impression server side was going to change to not return an error while still waiting (same as we don't for any other poll call). Can we get confirmation of server behavior on update poll beyond server long poll timeout? If anything, I am thinking this issue should _remove_ these error checks and only continue on non-error-non-outcome (this is a relic from the past where we, incorrectly IMO, returned error on poll timeout).

Reply:
I tested on the latest server RC and the server returns an empty outcome and no error

Reply:
:+1: I'd love to see us even work backwards in this PR and stop assuming that deadline exceeded (or any error) means continue to poll.